### ISSUE TITLE
flang driver reports version @FLANG_VERSION@

### ISSUE THREAD

**Issue by jprotze** [llvm/llvm-project#53908]
```
$ flang --version
flang version @FLANG_VERSION@
```

Expected result: flang should print the actual version number.
The script should be configured to allow cmake to replace the version number.

**Comment by banach-space** [comment]
Hi @jprotze , thanks for creating this issue!

That bash script is only there temporarily and I'm not aware of anyone developing it actively. Is there any chance that you could use the actual driver (`flang-new`) instead? That's where the main focus is ATM.

As for CMake, there's no support for LLVM Flang yet (see [here](https://gitlab.kitware.com/cmake/cmake/-/issues/22387) for CMake status). In fact, the script makes any interaction with CMake rather confusing - should CMake consider it as LLVM Flang or `gfortran`?

HTH,
-Andrzej

**Comment by jprotze** [comment]
Thanks for the hint to the new driver. 

Would it make sense to print a message from the script saying that flang-new should be used instead?
Having the pleasure to deal with Fortran code once a year, it's hard to follow, what the current name of the llvm Fortran driver is at the moment.

**Comment by banach-space** [comment]
>Would it make sense to print a message from the script saying that flang-new should be used instead?

It would, but there's a caveat. There's no upstream testing for this script (that would require access to `gfortran`). I will happily review and accept such change, but am a bit reluctant to submit one myself :) Having said that, I genuinely hope that we will be able to either retire or to rename this script on time for LLVM 15.

-Andrzej
